### PR TITLE
Remove column Deprecation

### PR DESCRIPTION
**Pull request by SeanArmstrong**
# Description

# TODO
- [x] Release notes
- [x] Documentation
- [x] Specs

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,43 @@
+# 2.0.0
+
+## Breaking Change
+
+### Removal of Row Deprecation
+
+* `Row`: can longer render any immediate children. A Column component has been introduced to maintain the column span, offset and align behaviour.
```
Comment (abramin) [inline CHANGELOG.md]: `no longer` ?

Comment (SeanArmstrong) [inline CHANGELOG.md]: :(

File: src/components/row/__snapshots__/__spec__.js.snap [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+exports[`Row columnClasses passes the prop to its children 1`] = `
+<div
+  className="carbon-row carbon-row--gutter-medium carbon-row--columns-0"
+/>
+`;
+
+exports[`Row columnDivide when columnDivide is true passes columnDivide to its children 1`] = `
+<div
+  className="carbon-row carbon-row--gutter-medium carbon-row--columns-0"
```
Comment (andrewjtait) [inline src/components/row/__snapshots__/__spec__.js.snap]: This snapshot is not testing what it is supposed to test (the column divide).

File: src/components/row/__snapshots__/__spec__.js.snap [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+</div>
+`;
+
+exports[`Row gutter applies a gutter class 1`] = `
+<div
+  className="carbon-row carbon-row--gutter-small carbon-row--columns-0"
+/>
+`;
+
+exports[`Row render renders a parent div with calculated CSS classes 1`] = `
```
Comment (andrewjtait) [inline src/components/row/__snapshots__/__spec__.js.snap]: this seems like it isn't testing anything in particular

File: src/components/row/__snapshots__/__spec__.js.snap [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Row columnClasses passes the prop to its children 1`] = `
+<div
+  className="carbon-row carbon-row--gutter-medium carbon-row--columns-0"
```
Comment (andrewjtait) [inline src/components/row/__snapshots__/__spec__.js.snap]: This snapshot is not testing what it is supposed to test (that columnClasses is passed down to the child column components - this example doesn't even have any child column components)

File: src/components/row/column/__snapshots__/__spec__.js.snap [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+exports[`Column options columnAlign renders a columnAlign class 1`] = `
+<div
+  className="carbon-column carbon-column--align-left"
+>
+  <span />
+</div>
+`;
+
+exports[`Column options columnDivide renders a columnOffset class 1`] = `
```
Comment (andrewjtait) [inline src/components/row/column/__snapshots__/__spec__.js.snap]: copy and paste error? should be columnDivide not columnOffset?

File: src/components/row/__snapshots__/__spec__.js.snap (hunk 3 above)
Comment (abramin) [inline src/components/row/__snapshots__/__spec__.js.snap]: It's the base case

**Pull request merged**